cobalt_compat.h

```diff
@@ -0,0 +1,115 @@
+#ifndef __NET_SCHED_COBALT_COMPAT_H
+#define __NET_SCHED_COBALT_COMPAT_H
+/* Backport some stuff if needed.
+ */
+#if KERNEL_VERSION(3, 14, 0) > LINUX_VERSION_CODE
+
+static inline u32 reciprocal_scale(u32 val, u32 ep_ro)
+{
+	return (u32)(((u64) val * ep_ro) >> 32);
+}
+
+#endif
+
+#if KERNEL_VERSION(3, 15, 0) > LINUX_VERSION_CODE
+
+static inline void kvfree(const void *addr)
+{
+	if (is_vmalloc_addr(addr))
+		vfree(addr);
+	else
+		kfree(addr);
+}
+
+#endif
+
+#if KERNEL_VERSION(3, 17, 0) > LINUX_VERSION_CODE
+
+#define ktime_get_ns() ktime_to_ns(ktime_get())
+
+#endif
+
+/* 3.18 > 4.7 use 3 arg, everything else uses 2 arg versions
+ * of qdisc_watchdog_schedule_ns
+ */
+#if ((KERNEL_VERSION(3, 18, 0) <= LINUX_VERSION_CODE) && (KERNEL_VERSION(4, 8, 0) > LINUX_VERSION_CODE))
+#define qdisc_watchdog_schedule_ns(_a, _b) qdisc_watchdog_schedule_ns(_a, _b, true);
+#endif
+
+#if KERNEL_VERSION(3, 18, 0) > LINUX_VERSION_CODE
+static inline void qdisc_qstats_backlog_dec(struct Qdisc *sch,
+					    const struct sk_buff *skb)
+{
+	sch->qstats.backlog -= qdisc_pkt_len(skb);
+}
+
+static inline void qdisc_qstats_backlog_inc(struct Qdisc *sch,
+					    const struct sk_buff *skb)
+{
+	sch->qstats.backlog += qdisc_pkt_len(skb);
+}
+
+static inline void __qdisc_qstats_drop(struct Qdisc *sch, int count)
+{
+	sch->qstats.drops += count;
+}
+
+static inline void qdisc_qstats_drop(struct Qdisc *sch)
+{
+	sch->qstats.drops++;
+}
+
+#endif
+
+#if KERNEL_VERSION(4, 1, 0) > LINUX_VERSION_CODE
+#if !defined(IS_REACHABLE)
```

Comment:
You can remove the !defined check if you are checking the kernel version anyway... Or leave it and drop the kernel version check, I guess...

Reply:
well it's a macro, so there shouldn't be any problems in just keeping the !defined check and dropping the kernel version check...